### PR TITLE
Hackfmt the entire repo

### PR DESCRIPTION
**Pull request by lexidor**
This PR formats all the files with hackfmt.
This makes it easier to collaborate with people using vscode-hack.
Vscode-hack will auto format on each save, causing accidental changes where unchanged code did not meet hackfmt's guides.
This enforces a habit of _Save without formatting_.
This is not only cumbersome, but it also adds to the code that is not hackfmt'ed.

### CODE REVIEW

**Comment by lexidor** [comment]
Running the test suite requires `vendor/bin/hacktest tests/BaseTest.hack tests/ComposerImporterTest.hack tests/ConfigurationLoaderTest.hack tests/FallbackHandlerTest.hack tests/IsDevTest.hack tests/RootImporterTest.hack tests/ScannerTest.hack` instead of `hackfmt tests`.
This explains why I could not run the tests locally.

File: src/ConfigurationLoader.hack [hunk 1]
```diff
@@ -34,10 +34,7 @@ abstract final class ConfigurationLoader {
       'Expected configuration file to contain a JSON object, got %s',
       \gettype($decoded),
     );
-    return self::fromData(
-      /* HH_IGNORE_ERROR[4110] */ $decoded,
-      $path,
-    );
+    return self::fromData(/* HH_IGNORE_ERROR[4110] */ $decoded, $path);
```
[the remark this answers is not in the document]
Comment (fredemmott) [reply src/ConfigurationLoader.hack]: if hackfmt did this, it's definitely a bug

**Comment by fredemmott** [comment]
Agreed with @jjergus - IMO those all should be raised as issues with hackfmt

**Comment by jjergus** [comment]
(note: this got merged in #42 which resolves the merge conflicts and avoids the HH_IGNORE_ERROR issue)

> those all should be raised as issues with hackfmt

Actually I'm not sure if those are even things that could be fixed, I was mostly just trying to make a general point about how sometimes a formatting decision made by a human can be better than strictly adhering to some hardcoded standard. But probably the pros of automatic formatting (effortless consistent formatting everywhere) outweigh the cons (a few suboptimally formatted lines here and there) so I don't mind it.

**Comment by fredemmott** [comment]
at least ignore_error/fixmes should not be changed; hackfmt already tries to special-case these, to avoid any changes, which often ends up with bad formatting (e.g. missing indents if you have two fixmes in the same trivia section), but that's what it's meant to prefer over any chance of a meaningful change

**Comment by jjergus** [comment]
Yeah that one makes sense to fix. For the other ones, I'm not really sure if we want complicated edge case rules like "don't add a trailing comma if the last argument is a method/pipe chain".

**Pull request closed without merging**